### PR TITLE
DOC: Test doctests in pandas/core/tools/datetimes.py 

### LINKED ISSUE #26401: Test doctests in pandas/core/tools/datetimes.py
#### Code Sample, a copy-pastable example if possible

Doctests in `to_datetime()`.

#### Problem description

It is being failed as below:

```
Failed example:
    df = pd.DataFrame({'year': [2015, 2016],
Exception raised:
    Traceback (most recent call last):
      File "/Applications/PyCharm.app/Contents/helpers/pycharm/docrunner.py", line 140, in __run
        compileflags, 1), test.globs)
      File "<doctest to_datetime[0]>", line 1
        df = pd.DataFrame({'year': [2015, 2016],
                                               ^
    SyntaxError: unexpected EOF while parsing
```

```
Failed example:
    pd.to_datetime([1, 2, 3], unit='D',
                   origin=pd.Timestamp('1960-01-01'))
Expected:
    0    1960-01-02
    1    1960-01-03
    2    1960-01-04
Got:
    DatetimeIndex(['1960-01-02', '1960-01-03', '1960-01-04'], dtype='datetime64[ns]', freq=None)
```

```
Failed example:
    %timeit pd.to_datetime(s,infer_datetime_format=False)
Exception raised:
    Traceback (most recent call last):
      File "/Applications/PyCharm.app/Contents/helpers/pycharm/docrunner.py", line 140, in __run
        compileflags, 1), test.globs)
      File "<doctest to_datetime[7]>", line 1
        %timeit pd.to_datetime(s,infer_datetime_format=False)
        ^
    SyntaxError: invalid syntax
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.3.final.0
python-bits: 64
OS: Darwin
OS-release: 18.5.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.2
pytest: 4.4.1
pip: 19.0.3
setuptools: 40.8.0
Cython: None
numpy: 1.16.2
scipy: None
pyarrow: 0.13.0
xarray: None
IPython: None
sphinx: 2.0.1
patsy: None
dateutil: 2.8.0
pytz: 2019.1
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml.etree: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10.1
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by HyukjinKwon**
- [x] closes #26401
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/tools/datetimes.py [hunk 1]
```diff
@@ -558,11 +557,16 @@ def to_datetime(arg, errors='raise', dayfirst=False, yearfirst=False,
...
-    100 loops, best of 3: 10.4 ms per loop
+    >>> import timeit
+    >>> timeit.timeit(
+    ...    lambda: pd.to_datetime(s, infer_datetime_format=True),
+    ...    number = 1)  # doctest: +SKIP
+    0.007785393000000029
 
-    >>> %timeit pd.to_datetime(s,infer_datetime_format=False)
-    1 loop, best of 3: 471 ms per loop
+    >>> timeit.timeit(
```
Comment (HyukjinKwon) [inline pandas/core/tools/datetimes.py]: I am not sure if it's fine to just convert to native Python compatible ones.

Comment (jorisvandenbossche) [reply pandas/core/tools/datetimes.py]: I think I personally prefer to keep the `%timeit` (which is easier to use / better practice IMO). 
Another option is to add a `# doctest: +SKIP` flag for those two

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks a lot!

File: pandas/core/tools/datetimes.py (hunk 1 above)
Comment (HyukjinKwon) [reply pandas/core/tools/datetimes.py]: Yea .. I prefer that too but `%timeit` is specific to IPython ....

Comment (jorisvandenbossche) [reply pandas/core/tools/datetimes.py]: Yes, but I think we can be opinionated about that. Maybe add a note between brackets before the example that the following only works in IPython?)

Comment (HyukjinKwon) [reply pandas/core/tools/datetimes.py]: okie!

Comment (HyukjinKwon) [reply pandas/core/tools/datetimes.py]: Actually, let me just don't add the note and revert it back with `# doctest: +SKIP`. I am sure people would know `%timeit` is for IPython ..

**Comment by jreback** [comment]
thanks @HyukjinKwon

**Comment by HyukjinKwon** [comment]
Thank you @jorisvandenbossche and @jreback

**Pull request merged**